lib/shared/media_view.dart

```diff
@@ -37,79 +40,101 @@ class MediaView extends StatelessWidget {
+                              borderRadius: BorderRadius.circular(6), // Image border
+                              child: Stack(
+                                alignment: Alignment.bottomRight,
+                                fit: StackFit.passthrough,
                                 children: [
-                                  const Padding(
-                                    padding: EdgeInsets.symmetric(horizontal: 8.0),
-                                    child: Icon(
-                                      Icons.link,
-                                      color: Colors.white60,
-                                    ),
-                                  ),
-                                  Expanded(
-                                    child: Text(
-                                      post?.url ?? '',
-                                      overflow: TextOverflow.ellipsis,
-                                      style: theme.textTheme.bodyMedium!.copyWith(
-                                        color: Colors.white60,
-                                      ),
+                                  Container(
+                                    color: Colors.grey.shade900,
+                                    padding: const EdgeInsets.symmetric(horizontal: 8.0, vertical: 12.0),
+                                    child: Row(
+                                      children: [
+                                        const Padding(
+                                          padding: EdgeInsets.symmetric(horizontal: 8.0),
+                                          child: Icon(
+                                            Icons.link,
+                                            color: Colors.white60,
+                                          ),
+                                        ),
+                                        Expanded(
+                                          child: Text(
+                                            post?.url ?? '',
+                                            overflow: TextOverflow.ellipsis,
+                                            style: theme.textTheme.bodyMedium!.copyWith(
+                                              color: Colors.white60,
+                                            ),
+                                          ),
+                                        ),
+                                      ],
                                     ),
                                   ),
                                 ],
                               ),
                             ),
-                          ],
+                            onTap: () => _launchURL(post?.url!),
+                          ),
                         ),
                       ),
-                      onTap: () => _launchURL(post?.url!),
                     ),
                   ),
-                ),
+                  Container(alignment: Alignment.center,
```

Comment:
I think for this one (feel free to let me know your own thoughts on it), but maybe it have the conditional first rather than within the `child` parameter

```
if (hideNsfw) Container(alignment: Aligntment.center ...)
```

This way, we don't have to render an empty Container, which might be better performance wise

Reply:
You're right, definitely an oversight on my part 😅

Fixed!